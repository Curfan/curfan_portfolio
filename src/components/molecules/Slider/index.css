.slider {
	width: 100%;
}

.default-slide {
	& > img {
		width: 100%;
	}
}
