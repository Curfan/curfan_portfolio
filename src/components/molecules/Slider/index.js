import React, { useEffect, useState, useImperativeHandle } from 'react';
import classnames from 'classnames';
import Swiper from 'react-id-swiper';
import 'react-id-swiper/lib/styles/css/swiper.css';

import styles from './index.css';

const DefaultSlideComponent = ({ className, image = '' }) => (
	<div className={classnames(styles.defaultSlide, className)}>
		<img src={image} alt="slide" />
	</div>
);

const Slider = React.forwardRef(
	(
		{
			className,
			slideStyle,
			slides = [],
			SlideComponent = DefaultSlideComponent,
			slideProps,
			config = {},
		},
		ref,
	) => {
		const [swiper, updateSwiper] = useState(null);
		const swiperConfig = { ...config };

		useEffect(() => {
			if (swiper) {
				swiper.slideTo(0);
				swiper.update();
			}
		}, [slides]);

		useImperativeHandle(ref, () => ({
			slideTo: props => {
				swiper.slideTo(props);
			},
		}));

		return (
			<div className={classnames(styles.slider, className)}>
				<Swiper getSwiper={updateSwiper} {...swiperConfig}>
					{slides.map(slide => (
						<SlideComponent key={slide.id} className={slideStyle} {...slide} {...slideProps} />
					))}
				</Swiper>
			</div>
		);
	},
);

// const Slider = ({
// 	className,
// 	slideStyle,
// 	slides = [],
// 	SlideComponent = DefaultSlideComponent,
// 	slideProps,
// 	config = {},
// }) => {
// 	const [swiper, updateSwiper] = useState(null);
// 	const swiperConfig = { ...config };

// 	useEffect(() => {
// 		if (swiper) {
// 			swiper.slideTo(0);
// 			swiper.update();
// 		}
// 	}, [slides]);

// 	return (
// 		<div className={classnames(styles.slider, className)}>
// 			<Swiper getSwiper={updateSwiper} {...swiperConfig}>
// 				{slides.map(slide => (
// 					<SlideComponent key={slide.id} className={slideStyle} {...slide} {...slideProps} />
// 				))}
// 			</Swiper>
// 		</div>
// 	);
// };

export default Slider;
