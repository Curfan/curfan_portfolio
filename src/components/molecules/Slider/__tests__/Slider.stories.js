import React from 'react';
import { storiesOf } from '@storybook/react';

import Slider from 'components/molecules/Slider';

const stories = storiesOf('molecules/Slider', module);

stories.add('__interactive', () => (
	<Slider
		slides={[
			{ id: 1, title: 'UI/UX 1', description: 'Task', image: 'https://fakeimg.pl/300/?text=01' },
			{ id: 2, title: 'UI/UX 2', description: 'Task', image: 'https://fakeimg.pl/300/?text=02' },
			{ id: 3, title: 'UI/UX 3', description: 'Task', image: 'https://fakeimg.pl/300/?text=03' },
			{ id: 4, title: 'UI/UX 4', description: 'Task', image: 'https://fakeimg.pl/300/?text=04' },
			{ id: 5, title: 'UI/UX 5', description: 'Task', image: 'https://fakeimg.pl/300/?text=05' },
		]}
		config={{ spaceBetween: 20, slidesPerView: 4 }}
	/>
));
