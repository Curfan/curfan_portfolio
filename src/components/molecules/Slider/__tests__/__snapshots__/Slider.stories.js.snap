// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="slider"
  >
    <div
      class="swiper-container swiper-container-initialized swiper-container-horizontal"
    >
      <div
        class="swiper-wrapper"
        style="-webkit-transition-duration: 0ms; -webkit-transform: translate3d(NaNpx, 0px, 0px);"
      >
        <div
          class="defaultSlide swiper-slide"
        >
          <img
            alt="slide"
            src="https://fakeimg.pl/300/?text=01"
          />
        </div>
        <div
          class="defaultSlide swiper-slide"
        >
          <img
            alt="slide"
            src="https://fakeimg.pl/300/?text=02"
          />
        </div>
        <div
          class="defaultSlide swiper-slide"
        >
          <img
            alt="slide"
            src="https://fakeimg.pl/300/?text=03"
          />
        </div>
        <div
          class="defaultSlide swiper-slide"
        >
          <img
            alt="slide"
            src="https://fakeimg.pl/300/?text=04"
          />
        </div>
        <div
          class="defaultSlide swiper-slide"
        >
          <img
            alt="slide"
            src="https://fakeimg.pl/300/?text=05"
          />
        </div>
      </div>
      <span
        aria-atomic="true"
        aria-live="assertive"
        class="swiper-notification"
      />
    </div>
  </div>
</DocumentFragment>
`;
